### ISSUE TITLE
Support Generics style PHPDoc @return typing

### ISSUE THREAD

**Issue by oojacoboo** [thecodingmachine/graphqlite#436]
When returning a Collection, you can annotate the `@return` type like:

```php
/**
 * @return Article[]
 */
#[GraphQLite\Field]
public function getArticles(): Collection; 
```

However, this doesn't tell many static analyzers that the return type is a Collection.  Maybe some smarter analyzers will check the actual return type, if typed, and use both combined.  I'm unsure about that.  But, there is value in using the Generics syntax for those that don't.

```php
/**
 * @return Collection<Article>
 */
#[GraphQLite\Field]
public function getArticles(): Collection;
```

This syntax is quickly becoming somewhat of a standard at this point.  It's supported by Psalm, Phan, PHPStan and I'm sure others.  It was [implemented by PHPStorm](https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-beta/).

Many of you are probably familiar with this syntax and probably use it.  For compatibility reasons, we should support both of these syntaxes when evaluating an `@return` annotation for the sake of generating GraphQL Types.  Currently, an error is thrown as it tries to create a Type definition out of `Collection`

**Comment by Lappihuan** [comment]
wouldn't #321 fix this?

**Comment by oojacoboo** [comment]
@Lappihuan yes, probably so - we need tests there to confirm.

**Comment by oojacoboo** [comment]
Generics support was added with #468 and released with version 6.0
